x/ibc-hooks/types/keys.go

```diff
@@ -0,0 +1,6 @@
+package types
+
+const (
+	ModuleName = "ibchooks"
+	StoreKey   = "hooks-for-ibc" // not using the module name because of collisions with key "ibc"
```

Comment:
oh interesting, whats the collission issue?

Reply:
I think this only happens when you have params (which this module doesn't have anymore), but I've seen conflicts when the key shares a prefix (rate limiting also has an inverted key name). I can double check the exact case where this matters

Reply:
The error if you use `ibchooks` is: 

```
    suite.go:77: test panicked: Potential key collision between KVStores:ibchooks - ibc
        goroutine 16 [running]:
        runtime/debug.Stack()
        	/opt/homebrew/Cellar/go/1.19/libexec/src/runtime/debug/stack.go:24 +0x64
        github.com/stretchr/testify/suite.failOnPanic(0x14000f4dba0, {0x102c86580, 0x14000fab740})
        	/Users/nicolas/go/pkg/mod/github.com/stretchr/testify@v1.8.1/suite/suite.go:77 +0x34
        github.com/stretchr/testify/suite.Run.func1.1()
        	/Users/nicolas/go/pkg/mod/github.com/stretchr/testify@v1.8.1/suite/suite.go:161 +0x23c
        panic({0x102c86580, 0x14000fab740})
        	/opt/homebrew/Cellar/go/1.19/libexec/src/runtime/panic.go:890 +0x26c
        github.com/cosmos/cosmos-sdk/types.assertNoPrefix({0x14000fca000, 0x1b, 0x1b})
        	/Users/nicolas/go/pkg/mod/github.com/osmosis-labs/cosmos-sdk@v0.45.1-0.20221118211718-545aed73e94e/types/store.go:91 +0x2d4
        github.com/cosmos/cosmos-sdk/types.NewKVStoreKeys({0x14000fca000, 0x1b, 0x1b})
        	/Users/nicolas/go/pkg/mod/github.com/osmosis-labs/cosmos-sdk@v0.45.1-0.20221118211718-545aed73e94e/types/store.go:105 +0x3c
        github.com/osmosis-labs/osmosis/v13/app/keepers.(*AppKeepers).GenerateKeys(0x14000fa6408)
        	/Users/nicolas/devel/osmosis/app/keepers/keys.go:15 +0x34
        github.com/osmosis-labs/osmosis/v13/app/keepers.(*AppKeepers).InitSpecialKeepers(0x14000fa6408, {0x102fd4b38, 0x14000f7c6d0}, 0x14000fba1a0, {0x14000f9bb80, 0x1d}, 0x14000010738, 0x5, 0x14000f7e9f0, {0x14000149ab8, ...})
        	/Users/nicolas/devel/osmosis/app/keepers/keepers.go:471 +0x54
        github.com/osmosis-labs/osmosis/v13/app.NewOsmosisApp({0x102fc4738, 0x103eda420}, {0x102fd3138, 0x14000f80220}, {0x0, 0x0}, 0x1, 0x14000f7e9f0, {0x14000149ab8, 0x18}, ...)
        	/Users/nicolas/devel/osmosis/app/app.go:187 +0x3ac
        github.com/osmosis-labs/osmosis/v13/app.Setup(0x0)
        	/Users/nicolas/devel/osmosis/app/test_helpers.go:32 +0xe0
        github.com/osmosis-labs/osmosis/v13/app/apptesting.(*KeeperTestHelper).Setup(0x14000dd0dc0)
        	/Users/nicolas/devel/osmosis/app/apptesting/test_suite.go:56 +0x2c
        github.com/osmosis-labs/osmosis/v13/x/ibc-hooks_test.(*HooksTestSuite).SetupTest(0x14000dd0dc0)
        	/Users/nicolas/devel/osmosis/x/ibc-hooks/ibc_middleware_test.go:40 +0x2c
        github.com/stretchr/testify/suite.Run.func1(0x14000f4dba0)
        	/Users/nicolas/go/pkg/mod/github.com/stretchr/testify@v1.8.1/suite/suite.go:165 +0x264
        testing.tRunner(0x14000f4dba0, 0x14000619b90)
        	/opt/homebrew/Cellar/go/1.19/libexec/src/testing/testing.go:1446 +0x174
        created by testing.(*T).Run
        	/opt/homebrew/Cellar/go/1.19/libexec/src/testing/testing.go:1493 +0x540
```

Reply:
oh wow, thats cool